app/models/panoptic/failed_job.rb

```diff
@@ -20,5 +20,13 @@ def exception_message
     def stacktrace
       failed_execution.error["backtrace"]
     end
+
+    def retry
+      failed_execution.retry
+    end
+
+    def discard
+      failed_execution.discard
+    end
```

Comment:
[Those two methods already exist in `SolidQueue::Job`](https://github.com/basecamp/solid_queue/blob/4a18d52d90a9e5bca06538b0023ec58c1426af92/app/models/solid_queue/job/executable.rb#L92-L98)

Plus the `retry` ensures safe operation on `failed_execution` in case it does not exist.

Reply:
Ahh, good catch. It totally slipped my mind that this was a subclass.